lib/happo/public/HappoDiffs.jsx

```diff
@@ -116,58 +116,116 @@ NewImages.propTypes = {
   images: PropTypes.arrayOf(PropTypes.shape(imageShape)).isRequired,
 };
 
+function maxImageSize(...imageUrls) {
+  const dimensions = {};
+
+  return new Promise((resolve, reject) => {
+    imageUrls.forEach((url, i) => {
+      const image = new Image();
+
+      image.onerrer = function handleImageError(e) {
+        reject(e);
+      };
+
+      image.onload = function handleImageLoad() {
+        const { width, height } = this;
+
+        // Use the index in case the URL is somehow duplicated.
+        dimensions[i] = { width, height };
+
+        if (Object.keys(dimensions).length >= imageUrls.length) {
+          // We are done, so compute the max width and height and resolve.
+          const values = Object.keys(dimensions).map(key => dimensions[key]);
+          const maxWidth = Math.max(...values.map(value => value.width));
+          const maxHeight = Math.max(...values.map(value => value.height));
+          resolve({ width: maxWidth, height: maxHeight });
+        }
+      };
+
+      image.src = url;
+    });
+  });
+}
+
 class Swiper extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      cursorLeftPercent: 50, // Start in the center
+      cursorLeft: 0,
+      height: 'auto',
+      width: 'auto',
     };
     this.handleMouseMove = this.handleMouseMove.bind(this);
   }
 
-  handleMouseMove(event) {
-    const leftPx = event.pageX - event.target.offsetLeft;
-    const leftPercent = (leftPx / event.target.offsetWidth) * 100;
+  componentWillMount() {
+    this.updateSize(this.props)
+      .then(({ width }) => {
+        // Start in the center
+        this.setState({ cursorLeft: width / 2 });
+      });
+  }
 
+  componentWillReceiveProps(nextProps) {
+    this.updateSize(nextProps);
+  }
```

Comment:
We don't right now, but I wanted this to work regardless and it was easy enough to do.